plugins/tiles/src/main/java/org/apache/struts2/tiles/StrutsApplicationResource.java

```diff
@@ -23,14 +23,41 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
+import java.net.MalformedURLException;
 import java.net.URL;
 
 public class StrutsApplicationResource extends PostfixedApplicationResource {
 
     private final URL url;
 
-    public StrutsApplicationResource(URL url) {
-        super(url.getPath());
+    /**
+     * workarounds WW-5011
+     */
+    private static String getExistedPath(URL url) throws MalformedURLException {
+        String path;
+        try {
+            //workarounds WW-5011 because includes ref in path
+            path = url.toURI().getPath();
+        } catch (Exception e) {
+            // fallback solution
+            path = url.getPath();
+
+            if (url.getRef() != null && !new File(path).exists()) {
```

Comment:
But getLastModified method already considers it should be a file.

Reply:
It seems `getLastModified()` considers the _possibility_ that resource (_path_) is a file, but _it doesn't assume it has to be_ a file.

If the path doesn't exist as a file (e.g. nonexistent file or non-file URL) then `exists()` will be false and `getLastModified()`  just returns 0. (As a side note the base Tiles interface also allows for an `IOException` to be thrown if there's no modified date available ... but Struts has attempted to avoid that).

Reply:
fixed :) - learnt from tiles URLApplicationResource.